### PR TITLE
tools, readme: update DM docs and add troubleshooting

### PR DESCRIPTION
**Pull request by lilin90**
@GregoryIan @CaitinChen PTAL
Via: https://github.com/pingcap/tidb-tools/pull/123

### CODE REVIEW

**Comment by GregoryIan** [comment]
@csuzhangxc PTAL

File: tools/data-migration-overview.md [hunk 1]
```diff
@@ -6,32 +6,38 @@ category: tools
 
 # Data Migration Overview
 
-Data Migration (DM) is an integrated scheduling and management tool based on mydumper, Loader, and Syncer. You can use DM to synchronize the full data or the incremental data from MySQL to TiDB.
+Data Migration (DM) is an integrated data synchronization task management platform that supports the full backup and the incremental synchronization of MariaDB/MySQL binlog into TiDB. It can help to reduce the operations cost and simplify the troubleshooting process.
```
Comment (GregoryIan) [inline tools/data-migration-overview.md]: How about `the full backup/restore and replicate MariaDB/MySQL binlog to TiDB`?

File: tools/data-migration-troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+However, in some cases, you need to [reset the data synchronization task](#reset-the-data-synchronization-task).
+
+## Common errors
+
+### `Access denied for user 'root'@'172.31.43.27' (using password: YES)` shows when you query the task or check the log
+
+For database related passwords in all the DM configuration files, use the passwords encrypted by `dmctl`. If a database password is empty, it is unnecessary to encrypt it. For how to encrypt the plaintext password, see [Encrypt the upstream MySQL user password using dmctl](../tools/data-migration-deployment.md#encrypt-the-upstream-mysql-user-password-using-dmctl).
+
+In addition, the user of the upstream and downstream databases must have the corresponding read and write privileges. Data Migration also checks some of the privileges automatically while starting the data synchronization task.
```
Comment (GregoryIan) [inline tools/data-migration-troubleshooting.md]: clarify what privileges needed

File: tools/data-migration-troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+When you manually handle the SQL statement that has an error, the frequently used commands include `query-status`, `sql-skip`, and `sql-replace`.
+
+### The manual process of handling abnormal SQL statements
+
+1. Use `query-status` to query the current running status of the task.
+
+    - Whether an error caused the `Paused` status of a task in a dm-worker
+    - Whether the cause of the task error is an error in executing the SQL statement
+
+2. Record the returned binlog pos (`SyncerBinlog`) that Syncer has synchronized when executing `query-status`.
```
[the remark this answers is not in the document]
Comment (GregoryIan) [reply tools/data-migration-troubleshooting.md]: We would provide a simple English document to customer tomorrow morning, just let them able to run DM, complete document would  be provided later

**Comment by GregoryIan** [comment]
@morgo PTAL

File: tools/data-migration-overview.md (hunk 1 above)
Comment (shenli) [inline tools/data-migration-overview.md]: the full backup -> full data migration and incremental data migration

File: tools/data-migration-overview.md [hunk 4]
```diff
@@ -6,32 +6,38 @@ category: tools
...
 
 ## Architecture
 
+The Data Migration tool consists of three components: dm-master, dm-worker, and dmctl.
```
Comment (shenli) [inline tools/data-migration-overview.md]: consists of -> includes

File: tools/data-migration-overview.md [hunk 5]
```diff
@@ -6,32 +6,38 @@ category: tools
...
+- Monitoring the running state of the data synchronization subtasks
 
-    - Persisting the binlog data to the local storage
-    - Storing the configuration information of the data synchronization subtasks
-    - Orchestrating the operation of the data synchronization subtasks
-    - Monitoring the running state of the data synchronization subtasks
+### dmctl 
 
-+ dmctl:
+dmctl is the access entry to the DM cluster.
```
Comment (shenli) [inline tools/data-migration-overview.md]: access entry  -> command line tool to control

**Comment by GregoryIan** [comment]
LGTM

**Comment by CaitinChen** [comment]
LGTM

File: tools/data-migration-troubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+2. If the error you encountered is not involved yet, and you cannot solve the problem yourself by checking the log or monitoring metrics, you can contact the corresponding sales support staff.
+
+3. After the error is solved, restart the task using dmctl.
+
+    ```bash
+    resume-task ${task name}
+    ```
+
+However, in some cases, you need to [reset the data synchronization task](#reset-the-data-synchronization-task).
```
Comment (shenli) [inline tools/data-migration-troubleshooting.md]: "in some cases"  Could you make it more specific?

Comment (lilin90) [reply tools/data-migration-troubleshooting.md]: The link here links to the specific details. I'll update the wording here in another PR.

**Pull request merged**